src/internal/connector/exchange/attachment_uploadable.go

```diff
@@ -0,0 +1,139 @@
+package exchange
+
+import (
+	"context"
+
+	"github.com/microsoftgraph/msgraph-sdk-go/models"
+	ups "github.com/microsoftgraph/msgraph-sdk-go/users/item/calendars/item/events/item/attachments/createuploadsession"
+	"github.com/microsoftgraph/msgraph-sdk-go/users/item/messages/item/attachments/createuploadsession"
+	"github.com/pkg/errors"
+
+	"github.com/alcionai/corso/src/internal/connector/graph"
+	"github.com/alcionai/corso/src/internal/connector/support"
+)
+
+// attachementUploadable represents structs that are able to upload small attachments directly to an item or use an
+// upload session to connect large attachments to their corresponding M365 item.
+type attachmentUploadable interface {
+	uploadSmallAttachment(ctx context.Context, attachment models.Attachmentable) error
+	uploadSession(ctx context.Context, attachName string, attachSize int64) (models.UploadSessionable, error)
+	// getItemID returns the M365ID of the item associated with  the attachment
+	getItemID() string
+}
+
+var (
+	_ attachmentUploadable = &mailAttachmentUploader{}
+	_ attachmentUploadable = &eventAttachmentUploader{}
+)
+
+// mailAttachmentUploader is a struct that is able to upload attachments for exchange.Mail objects
+type mailAttachmentUploader struct {
+	userID   string
+	folderID string
+	itemID   string
+	service  graph.Service
+}
+
+func (mau *mailAttachmentUploader) getItemID() string {
+	return mau.itemID
+}
+
+func (mau *mailAttachmentUploader) uploadSmallAttachment(ctx context.Context, attach models.Attachmentable) error {
+	_, err := mau.service.Client().
+		UsersById(mau.userID).
+		MailFoldersById(mau.folderID).
+		MessagesById(mau.itemID).
+		Attachments().
+		Post(ctx, attach, nil)
+	if err != nil {
+		return errors.Wrap(err, support.ConnectorStackErrorTrace(err))
+	}
+
+	return nil
+}
+
+func (mau *mailAttachmentUploader) uploadSession(
+	ctx context.Context,
+	attachmentName string,
+	attachmentSize int64,
+) (models.UploadSessionable, error) {
+	session := createuploadsession.NewCreateUploadSessionPostRequestBody()
+
+	attItem := models.NewAttachmentItem()
+	attType := models.FILE_ATTACHMENTTYPE
+	attItem.SetAttachmentType(&attType)
+	attItem.SetName(&attachmentName)
+	attItem.SetSize(&attachmentSize)
```

Comment:
nit: this is the same for both emails and events. Can you please make it a separate helper function in this file that just creates and returns the attachment with the proper information? That helps keep things consistent and easy to update as there's a single place to change stuff